txtorcon/torcontrolprotocol.py

```diff
@@ -103,6 +104,25 @@ def unlisten(self, cb):
 
     def got_update(self, data):
         # print self.name,"got_update:",data
+        if self.use_stem:
+            import stem.response
+            response_data = ""
+            # Add appropriate headers for parsing into stem.response.ControlMessage instance
+            if self.name == "CIRC":
+                response_data = "650 " + self.name + " " + data.replace('\n', ' ') + "\r\n"
+            elif self.name == "NEWCONSENSUS" or self.name == "NS":
+                lines = data.split('\n')
+                response_data = "650+" + self.name + "\r\n"
+                for line in lines[:-1]:
+                    response_data += line + "\r\n"
+                response_data += ".\r\n650 OK\r\n"
+            else:
+                response_data = "650 " + self.name + " " + data + "\r\n"
+            try:
+                response_data = stem.response.ControlMessage.from_str(response_data)
+                stem.response.convert('EVENT', response_data)
+            except:
+                pass
         for cb in self.callbacks:
             cb(data)
 
```

Comment:
Yes, currently it just calls data. If the rest of the changes look good, then I will do the same to get back the required string to pass into cb()